crates/languages/src/rust.rs

```diff
@@ -510,17 +510,23 @@ impl ContextProvider for RustContextProvider {
         cx: &App,
     ) -> Option<TaskTemplates> {
         const DEFAULT_RUN_NAME_STR: &str = "RUST_DEFAULT_PACKAGE_RUN";
+        const CUSTOM_TARGET_DIR: &str = "RUST_TARGET_DIR";
         let package_to_run = language_settings(Some("Rust".into()), file.as_ref(), cx)
             .tasks
             .variables
             .get(DEFAULT_RUN_NAME_STR)
             .cloned();
+        let custom_target_dir = language_settings(Some("Rust".into()), file.as_ref(), cx)
```

Comment:
Let's not call `language_settings` twice here, as it's kinda expensive; could you extract that to a separate temporary and use it in both `package_to_run` and `custom_target_dir`?